src/components/app/details/appConfig/AppConfig.utils.ts

```diff
@@ -3,59 +3,78 @@ import { DOCUMENTATION } from '../../../../config'
 import { AppStageUnlockedType, STAGE_NAME } from './appConfig.type'
 
 //stage: last configured stage
-export const isUnlocked = (stage: string): AppStageUnlockedType => {
+export const isUnlocked = (stage: string, isGitOpsConfigurationRequired?: boolean): AppStageUnlockedType => {
     return {
         material:
             stage === STAGE_NAME.APP ||
             stage === STAGE_NAME.GIT_MATERIAL ||
             stage === STAGE_NAME.CI_CONFIG ||
             stage === STAGE_NAME.CI_PIPELINE ||
             stage === STAGE_NAME.DEPLOYMENT_TEMPLATE ||
+            stage === STAGE_NAME.GITOPS_CONFIG ||
             stage === STAGE_NAME.CD_PIPELINE ||
             stage === STAGE_NAME.CHART_ENV_CONFIG,
         dockerBuildConfig:
             stage === STAGE_NAME.GIT_MATERIAL ||
             stage === STAGE_NAME.CI_CONFIG ||
             stage === STAGE_NAME.CI_PIPELINE ||
             stage === STAGE_NAME.DEPLOYMENT_TEMPLATE ||
+            stage === STAGE_NAME.GITOPS_CONFIG ||
             stage === STAGE_NAME.CD_PIPELINE ||
             stage === STAGE_NAME.CHART_ENV_CONFIG,
         deploymentTemplate:
             stage === STAGE_NAME.CI_CONFIG ||
             stage === STAGE_NAME.CI_PIPELINE ||
             stage === STAGE_NAME.DEPLOYMENT_TEMPLATE ||
+            stage === STAGE_NAME.GITOPS_CONFIG ||
             stage === STAGE_NAME.CD_PIPELINE ||
             stage === STAGE_NAME.CHART_ENV_CONFIG,
-        workflowEditor:
+        gitOpsConfig:
             stage === STAGE_NAME.CI_PIPELINE ||
             stage === STAGE_NAME.DEPLOYMENT_TEMPLATE ||
+            stage === STAGE_NAME.GITOPS_CONFIG ||
+            stage === STAGE_NAME.CD_PIPELINE ||
+            stage === STAGE_NAME.CHART_ENV_CONFIG,
+        workflowEditor:
+            stage === STAGE_NAME.CI_PIPELINE ||
+            (isGitOpsConfigurationRequired
```

Comment:
we can common out this to reuse

Reply:
Used const isCommonUnlocked variable to reduce duplicity